vividus-extension-selenium/src/main/java/org/vividus/selenium/screenshot/AbstractAshotFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.selenium.screenshot;
+
+import static ru.yandex.qatools.ashot.shooting.ShootingStrategies.cutting;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BinaryOperator;
+import java.util.function.Supplier;
+
+import javax.inject.Inject;
+
+import com.google.common.base.Suppliers;
+
+import org.apache.commons.lang3.Validate;
+import org.vividus.selenium.screenshot.strategies.ScreenshotShootingStrategy;
+import org.vividus.util.property.PropertyMappedCollection;
+
+import ru.yandex.qatools.ashot.shooting.ShootingStrategies;
+import ru.yandex.qatools.ashot.shooting.ShootingStrategy;
+import ru.yandex.qatools.ashot.shooting.cutter.FixedCutStrategy;
+
+public abstract class AbstractAshotFactory<T extends ScreenshotConfiguration> implements IAshotFactory<T>
+{
+    private PropertyMappedCollection<T> ashotConfigurations;
+    private String shootingStrategy;
+    private Map<String, ScreenshotShootingStrategy> strategies;
+    private final Supplier<ShootingStrategy> baseShootingStrategyProvider = Suppliers
+            .memoize(() -> ShootingStrategies.scaling(this.getDpr()));
+
+    protected Optional<T> getAshotConfiguration(Optional<T> ashotConfiguration, BinaryOperator<T> merger)
+    {
+        return ashotConfiguration.map(c -> merger.apply(c, ashotConfigurations.getNullable(shootingStrategy).get()))
+                                 .or(() -> ashotConfigurations.getNullable(shootingStrategy));
+    }
+
+    protected ShootingStrategy configureNativePartialsToCut(int headerToCut, T screenshotConfiguration,
+            ShootingStrategy decorated)
+    {
+        ShootingStrategy toDecorate = decorated;
+        int nativeFooterToCut = screenshotConfiguration.getNativeFooterToCut();
+        int nativeHeaderToCut = headerToCut;
+        if (anyNotZero(nativeFooterToCut, nativeHeaderToCut))
```

Comment:
is it really required?

Reply:
there will be unnecessary extra calls in case of 0, 0 per screenshot

Reply:
for example?

Reply:
https://github.com/pazone/ashot/blob/41a81919b40cc745d0652eae487e1f2ff54c5c41/src/main/java/pazone/ashot/CuttingDecorator.java#L45-L52

Reply:
what about updating `CuttingDecorator` to make it more smart? (maybe not for current PR)